components/Tabs/src/private/MDCItemBar.m

```diff
@@ -513,7 +513,10 @@ - (void)selectItemAtIndex:(NSUInteger)index animated:(BOOL)animated {
 }
 
 - (UITabBarItem *)itemAtIndexPath:(NSIndexPath *)indexPath {
-  return _items[indexPath.item];
+  if (_items && (NSInteger)_items.count > indexPath.item) {
+    return _items[indexPath.item];
+  }
+  return nil;
```

Comment:
Please change the receivers for this to handle `nullable` values.  Notably `MDCTabBarDelegate`'s `tabBar:didSelectItem:` cannot handle `nil`. Please review the rest and update them as necessary.

Reply:
A quick review shows that `MDCTabBarPrivateIndicatorContext` also cannot handle a `nil` item.

Reply:
Is it alright with you if this stuff happens in a follow up PR? I think we might do a hotfix for this, and I'd rather it only have the original change and the code that fixes it.

Reply:
It’s possible this fix may result in other issues due to the new behavior so I also recommend addressing the call sites that make use of this api.

Reply:
It’s possible this fix may result in other issues due to the new behavior so I also recommend addressing the call sites that make use of this api.

Reply:
This feels like doing more than addressing the thing that got the initial PR reverted in the first place, which feels against the spirit of hotfixes, but ok

Reply:
To be clear, this isn't a hotfix - no clients are blocked on this change. This is a bug fix of existing code that will unblock feature work. Although the feature work is important to an internal client, I don't believe it warrants a hotfix release of the library.

Reply:
There had been some talk about releasing it as a hotfix. Comments like [this one](https://github.com/material-components/material-components-ios/pull/7518#issuecomment-497508314) make it sound like clients are blocked on this change. I'm fine with not releasing it as a hotfix though, whatever works. I'll let Cody know, he was asking about it.